### PR TITLE
[CI] Updated drone.star file to match latest changes

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This is an update on drone.star file 
- using drone-cancel-previous-builds
- using thegeeklab/drone-github-comment instead of jmccann/drone-github-comment for github comments 
- moving docker image references to constants
- using caches images
- Do not run PHP unit tests with Oracle by default
- Skip pipelines based on changed code

## Related Issue
Part of: 
- https://github.com/owncloud/QA/issues/700
- https://github.com/owncloud/QA/issues/717
- https://github.com/owncloud/QA/issues/729
- https://github.com/owncloud/QA/issues/728
- https://github.com/owncloud/core/issues/38679
- https://github.com/owncloud/QA/issues/730

### CODE REVIEW

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/files_antivirus/2396
`linter: duplicate pipeline names`

File: .drone.star [hunk 1]
```diff
@@ -23,12 +50,12 @@ config = {
     "phpunit": {
         "allDatabases": {
             "phpVersions": [
-                "7.3",
+                DEFAULT_PHP_VERSION,
```
Comment (phil-davis) [inline .drone.star:53]: You can't run unit tests twice on the same DEFAULT_PHP_VERSION

File: .drone.star [hunk 2]
```diff
@@ -37,235 +64,225 @@ config = {
...
-            "emailNeeded": True,
-            "runCoreTests": True,
-            "federatedServerNeeded": True,
-            "cron": "nightly",
-            "runAllSuites": True,
-            "numberOfParts": 5,
-            "filterTags": "@smokeTest&&~@skip&&~@app-required",
-        },
-    },
+    # "acceptance": {
```
Comment (kiranparajuli589) [inline .drone.star:67]: what happened here?

File: .drone.star [hunk 3]
```diff
@@ -1005,6 +1021,9 @@ def phpTests(ctx, testType, withCoverage):
             for db in params["databases"]:
                 keyString = "-" + category if params["includeKeyInMatrixName"] else ""
                 name = "%s%s-php%s-%s" % (testType, keyString, phpVersion, db.replace(":", ""))
+                print(name)
+
+                # print(params["databases"])
```
Comment (kiranparajuli589) [inline .drone.star:1026]: debug lines?

**Review by kiranparajuli589 — CHANGES_REQUESTED** [review]

File: .drone.star (hunk 2 above)
Comment (phil-davis) [reply .drone.star:67]: @grgprarup please revert the various "debug" changes.

File: .drone.star (hunk 3 above)
Comment (phil-davis) [reply .drone.star]: removed

**Review by phil-davis — APPROVED** [review]

**Pull request merged**